src/test/scala/firrtlTests/InfoSpec.scala

```diff
@@ -172,4 +192,51 @@ class InfoSpec extends FirrtlFlatSpec with FirrtlMatchers {
     val expectedInfos = Seq(FileInfo(StringLit("Top.scala 15:14")), FileInfo(StringLit("myfile.fir 6:4")))
     circuitState should containTree { case MultiInfo(`expectedInfos`) => true }
   }
+
+  "source locators for basic register updates" should "be propagated to Verilog" in {
+    val result = compileBody(s"""
+      |input clock : Clock
+      |input reset : UInt<1>
+      |output io : { flip in : UInt<8>, out : UInt<8>}
+      |reg r : UInt<8>, clock
+      |r <= io.in $Info1
+      |io.out <= r $Info2
+      |""".stripMargin
+    )
+    result should containLine (s"r <= io_in; //$Info1")
+  }
+
+  "source locators for register reset" should "be propagated to Verilog" in {
+    val result = compileBody(s"""
+      |input clock : Clock
+      |input reset : UInt<1>
+      |output io : { flip in : UInt<8>, out : UInt<8>}
+      |reg r : UInt<8>, clock with : (reset => (reset, UInt<8>("h0"))) $Info3
+      |r <= io.in $Info1
+      |io.out <= r $Info2
```

Comment:
`Info2` isn't checked in the test. Should it be removed?

Reply:
Same as above, put differently, it's checking that `Info1` is prioritized over `Info2`, I should add a comment mentioning this.

Reply:
with another look I realized that this is different from the above, I'll remove it here.